Task2/domain/src/main/java/com/epam/esm/entity/GiftCertificate.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.esm.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+@Data
+public class GiftCertificate {
+    private long id;
```

Comment:
Easier to use with generics and other constructions that require objects, not primitives I guess. Also from the perspective of safety, it can be better as these types of values are stored in different memory parts(heap and stack) and it may result in fewer opportunities to somehow steal this data. I remember reading something about it, but not so sure if it really makes any difference